tests/e2e/map/map_shapefile.spec.js

```diff
@@ -0,0 +1,468 @@
+
+        // After clicking two shapes the map will have 6 path elements
+        await expect(await page.locator('g path').all()).toHaveLength(6)
+
+        // Checking that the right number of results are loaded ensures that the route
+        // was fulfilled correctly with the succesfull paramCheck
+        await expect(page.getByText('Showing 2 of 2 matching collections')).toBeVisible()
+
+        // Updates the URL
+        await expect(page).toHaveURL('search?polygon[0]=42.1875%2C-16.46517%2C56.25%2C-16.46517%2C42.1875%2C-2.40647%2C42.1875%2C-16.46517&polygon[1]=58.25%2C-14.46517%2C58.25%2C0.40647%2C44.1875%2C0.40647%2C58.25%2C-14.46517&sf=1&sfs[0]=0&sfs[1]=1&lat=-8.296765000000008&long=44.625&zoom=4')
+      })
+    })
+
+    test.describe('When the shapefile has a polygon with too many points', () => {
+      test('renders correctly', async ({ page }) => {
+        await interceptUnauthenticatedCollections({
+          page,
+          body: commonBody,
+          headers: commonHeaders,
+          additionalRequests: [{
+            body: tooManyPointsShapefileBody,
+            headers: {
+              ...commonHeaders,
+              'cmr-hits': '2'
+            },
+            paramCheck: (parsedQuery) => parsedQuery?.polygon?.[0] === '-114.04999,36.95777,-114.0506,37.0004,-114.04826,41.99381,-119.99917,41.99454,-120.00101,38.99957,-118.71431,38.10218,-117.50012,37.22038,-116.0936,36.15581,-114.63667,35.00881,-114.63689,35.02837,-114.60362,35.06423,-114.64435,35.1059,-114.57852,35.12875,-114.56924,35.18348,-114.60431,35.35358,-114.67764,35.48974,-114.65431,35.59759,-114.68941,35.65141,-114.68321,35.68939,-114.70531,35.71159,-114.69571,35.75599,-114.71211,35.80618,-114.67742,35.87473,-114.73116,35.94392,-114.74376,35.9851,-114.73043,36.03132,-114.75562,36.08717,-114.57203,36.15161,-114.51172,36.15096,-114.50217,36.1288,-114.45837,36.13859,-114.44661,36.12597,-114.40547,36.14737,-114.37211,36.14311,-114.30843,36.08244,-114.31403,36.05817,-114.25265,36.02019,-114.14819,36.02801,-114.11416,36.09698,-114.12086,36.1146,-114.09987,36.12165,-114.04684,36.19407,-114.04999,36.95777'
+          }]
+        })
+
+        await page.route(/shapefiles$/, async (route) => {
+          await route.fulfill({
+            json: { shapefile_id: '1' },
+            headers: { 'content-type': 'application/json; charset=utf-8' }
+          })
+        })
+
+        await page.goto('/')
+
+        // Upload the shapefile
+        await uploadShapefile(page, 'too_many_points.geojson')
+
+        // Waiting for the URL to include the correct zoom level ensures the map is finished drawing
+        await page.waitForURL(/zoom=6/, { timeout: 3000 })
+
+        await expect(page.getByTestId('filter-stack-item__hint')).toHaveText('1 shape selected')
+
+        // The shapefile has 1 svg shapes that is auto selected, so 2 path elements
+        await expect(await page.locator('g path').all()).toHaveLength(2)
+
+        // Displays a modal
```

Comment:
We were closing the modal in the old one  // Close the modal
        await page.getByRole('button', { name: 'Close' }).click()
        I don't it was necessary

Reply:
Right, I didn't see the point